File: StudentMnagement/src/main/java/com/adobe/StudentManagement/model/Marks.java
package com.adobe.StudentManagement.model;

import lombok.*;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.adobe.StudentManagement.config.ObjectIdSerializer;
import com.adobe.StudentManagement.config.ObjectIdDeserializer;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "marks")
public class Marks {

    @Id
    @JsonSerialize(using = ObjectIdSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private ObjectId id;

    @JsonSerialize(using = ObjectIdSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private ObjectId studentId;

    @JsonSerialize(using = ObjectIdSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private ObjectId subjectId;
    private int semester;
    private int marks;
}

File: StudentManagementNode/models/Marks.js
import mongoose from 'mongoose';

const marksSchema = new mongoose.Schema({
  studentId: { type: mongoose.Schema.Types.ObjectId, ref: 'Student', required: true },
  subjectId: { type: mongoose.Schema.Types.ObjectId, ref: 'Subject', required: true },
  semester: { type: Number, required: true },
  marks: { type: Number, required: true },
  
}, { timestamps: true });

const Marks = mongoose.model('Marks', marksSchema);
export default Marks;
